src/averageAges.js

```diff
@@ -14,12 +14,34 @@
  *
  * @return {number}
  */
+
 function calculateMenAverageAge(people, century) {
   // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => person.sex === 'm');
+
+  const ages = [];
+
+  men.map(function(person) {
+    ages.push(person.died - person.born);
+  });
+
+  const avWithoutCent = ages.reduce((a, b) => a + b, 0) / ages.length;
+
+  const withCent = men.filter((person) => {
+    return Math.ceil(person.died / 100) === century;
+  });
+
+  const agesWith = [];
+
+  withCent.map(function(person) {
+    agesWith.push(person.died - person.born);
+  });
+
+  const avWithCent = agesWith.reduce((a, b) => a + b, 0) / agesWith.length;
+
+  const result = (!century) ? avWithoutCent : avWithCent;
```

Comment:
by these variable names i am not able to understand what these variables are all about
and also instead of calculating all the values in different variables may be we can use a ternary operator and get the solution on 2 lines of code

Reply:
I have applied the same changes to the ret of the code, too